apps/openmw/mwworld/containerstore.cpp

```diff
@@ -241,7 +248,12 @@ bool MWWorld::ContainerStore::stacks(const ConstPtr& ptr1, const ConstPtr& ptr2)
     }
 
     return ptr1 != ptr2 // an item never stacks onto itself
-        && ptr1.getCellRef().getOwner() == ptr2.getCellRef().getOwner()
+        && (
+            ptr1.getCellRef().getOwner() == ptr2.getCellRef().getOwner()
```

Comment:
The `ContainerStore::stacks` checks if two items can be combined to the one CellRef with increased count of objects.
If these two items have different owners, one of them will be lost, but it does not matter with our current implementation (see #1941).
Issues will arise only if you aimed to keep original item owners to track stolen items, as you wanted.
In this case every item with different owner should be in the separate stack in the inventory, and we should combine them only in ItemStacks in item models. I tried to do it, but it does not work well for stackable eqipped items (such as throwing weapons and ammo).

Reply:
> Will it change save format?

Yes, it will.

> But still we do Morrowind.

Then we should decide if we want to keep an original system with its quirks or if we want to replace it later.
In the first case there is a little we can do here:
1. Merge stacks with different owners as in #1941.
2. Take in account Soul field when tracking stolen items (requires a savegame format change).
3. Add a workaround for evidence chests ([bug #3609](https://gitlab.com/OpenMW/openmw/issues/3609)).

Keeping the Owner field during item transfers is not a reliable approach here.